homeassistant/components/dexcom/config_flow.py

```diff
@@ -0,0 +1,71 @@
+"""Config flow for Dexcom integration."""
+import logging
+
+from pydexcom import AccountError, Dexcom, SessionError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_UNIT_OF_MEASUREMENT, CONF_USERNAME
+
+from .const import (  # pylint:disable=unused-import
+    CONF_SERVER,
+    DOMAIN,
+    MG_DL,
+    MMOL_L,
+    SERVER_OUS,
+    SERVER_US,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Optional(CONF_SERVER, default=SERVER_US): vol.In({SERVER_US, SERVER_OUS}),
+        vol.Optional(CONF_UNIT_OF_MEASUREMENT, default=MG_DL): vol.In({MG_DL, MMOL_L}),
```

Comment:
Optional options should be part of an options flow and not the main flow.
https://developers.home-assistant.io/docs/config_entries_options_flow_handler

Reply:
I wen't ahead and converted the optional options to required, as the integration will not function correctly if not provided correctly.

Reply:
Isn't unit of measurement still something that belongs in the options flow? It doesn't have anything to do with setting up the connection.

Items with default values don't need to be required regardless.

Reply:
Gotcha, using option flow for unit of measurement now!